### PR TITLE
[Authentication] fix bug and implement google oauth

### PR DESCRIPTION
**Pull request by okasato**
- Fix a bug in local login
- Implement google oauth
  - after finishing passport in backend, access token returns to frontend (App.js)
- Add userId in default state

### CODE REVIEW

File: src/components/Login.jsx [hunk 1]
```diff
@@ -48,10 +48,18 @@ class Login extends Component {
...
-    this.props.changePage('dashboard');
-    this.props.setIsLoggedIn();
+    signIn(nodeEmail.value, nodePsw.value)
+      .then(response => {
+        if (response.status !== 200) {
+          nodeErrorMessage.className = 'error-message';
+          nodeErrorMessage.innerHTML = 'Your password or email is wrong.';
+          return;
+        }
+        response.response.then(res => this.props.setUserId(res[0].id));
```
Comment (ChristianSchmidt1981) [inline src/components/Login.jsx]: rename res to response

Comment (ChristianSchmidt1981) [reply src/components/Login.jsx]: response.response.then is double response correct?

File: src/components/CreateCard.jsx [hunk 2]
```diff
@@ -50,13 +50,15 @@ class CreateCard extends Component {
...
+console.log(event.target.dataset);
     switch (event.target.dataset.linkedTo) {
       case 'frontText':
         this.setMicrophoneActive(1);
         hearingLanguageCode = 'ja-JP';
         callbackFunction = (text) => {
-          this.setMicrophoneInActive();
+          console.log(111, text);
           this.props.updateFrontText(text);
+          console.log(222);
```
Comment (ChristianSchmidt1981) [inline src/components/CreateCard.jsx]: delete console.log

File: src/components/CreateCard.jsx [hunk 3]
```diff
@@ -50,13 +50,15 @@ class CreateCard extends Component {
...
     let hearingLanguageCode = '';
     let callbackFunction;
+console.log(event.target.dataset);
     switch (event.target.dataset.linkedTo) {
       case 'frontText':
         this.setMicrophoneActive(1);
         hearingLanguageCode = 'ja-JP';
         callbackFunction = (text) => {
-          this.setMicrophoneInActive();
+          console.log(111, text);
```
Comment (ChristianSchmidt1981) [inline src/components/CreateCard.jsx]: delete console.log

File: src/components/CreateCard.jsx [hunk 4]
```diff
@@ -50,13 +50,15 @@ class CreateCard extends Component {
     const speech = new this.WebSpeechApi();
     let hearingLanguageCode = '';
     let callbackFunction;
+console.log(event.target.dataset);
```
Comment (ChristianSchmidt1981) [inline src/components/CreateCard.jsx]: delete console.log

File: server/routes/localLogin.js [hunk 5]
```diff
@@ -1,15 +1,19 @@
...
 
 router.post(
   '/login',
-  passport.authenticate('local', { failureRedirect: '/auth/login' }),
-  (request, response) => {
-    response.redirect('http://localhost:3000');
+  passport.authenticate('local'),
+  
+  async (request, response) => {
+    const user = await db.users.findByAccount(response.req.body.username);
```
Comment (ChristianSchmidt1981) [inline server/routes/localLogin.js:11]: response.req. is correct?

File: server/app.js [hunk 6]
```diff
@@ -19,23 +19,24 @@ app.use(bodyParser.json());
 app.use(bodyParser.urlencoded({ extended: false }));
 app.use(cookieParser());
 
+// app.use(express.static(path.join(__dirname, '../public')));
```
Comment (ChristianSchmidt1981) [inline server/app.js]: delete this line

File: server/passport/localLogin.js [hunk 7]
```diff
@@ -6,7 +6,7 @@ const db = require('../db');
 module.exports = new Strategy(async (username, password, done) => {
   try {
     const user = await db.users.findByAccount(username);
-    if (!user) {
+    if (!user.length) {
```
Comment (ChristianSchmidt1981) [inline server/passport/localLogin.js]: replace with 
if (user.length === 0) {

File: src/components/CreateCard.jsx [hunk 8]
```diff
@@ -140,46 +142,38 @@ class CreateCard extends Component {
...
-                  className="microphone"
-                  id="microphone-learning-2"
-                  data-linked-to="backText"
-                  data-id="2"
-                  onClick={this.handleRecordClick}
-                />
-              </div>
+              <strong>back text</strong>
+              <textarea
+                name="backText"
```
Comment (ChristianSchmidt1981) [inline src/components/CreateCard.jsx]: I've the feeling, that this changes (in this file) are not extra existing.
maybe a merge-conflict issue?

File: src/components/Top.jsx [hunk 9]
```diff
@@ -0,0 +1,29 @@
+import React, { Component } from 'react';
+import { 
+  BrowserRouter as Router,
+  Route, 
+  Link,
```
Comment (ChristianSchmidt1981) [inline src/components/Top.jsx]: did we need Link?

File: src/utils/isTokenValid.js [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+      Accept: 'application/json',
+    })
+  })
+  .then(response => response.json());
+}
+
+export {
+  googleOauth,
+  isTokenValid
+}
```
Comment (ChristianSchmidt1981) [inline src/utils/isTokenValid.js]: no newline at end of file.

please add an empty code-line

**Review by ChristianSchmidt1981 — CHANGES_REQUESTED** [review]
I've some smaller issues and some questions.

I think one file is a mistake from a merge-conflict

**Review by ryutaudo — APPROVED** [review]

**Review by ChristianSchmidt1981 — APPROVED** [review]

**Pull request merged**